### PR TITLE
ConsoleLoggerProvider should reuse ConsoleUtils helper to align the behaviors

### PR DESCRIPTION
**Pull request by adamsitnik**
A follow up to https://github.com/dotnet/runtime/pull/88241

Reuse the code so the behavior of System.Console and Microsoft.Extensions.Logging.Console are aligned. Microsoft.Extensions.Logging.Console is now also supporting `NO_COLOR` env var:

https://github.com/dotnet/runtime/blob/a3981dde2de089061fe7c857d8bd5da573546a3f/src/libraries/Common/src/System/Console/ConsoleUtils.cs#L34

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs [hunk 1]
```diff
@@ -68,32 +68,30 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IEnu
         [UnsupportedOSPlatformGuard("windows")]
         private static bool DoesConsoleSupportAnsi()
         {
-            string? envVar = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION");
-            if (envVar is not null && (envVar == "1" || envVar.Equals("true", StringComparison.OrdinalIgnoreCase)))
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs:72]: I am wondering what should be the behavior if someone set `DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION=1` and the output is not redirected? This env variable will be fully ignored at that time. is this the expected behavior?

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs:72]: CC @hach-que

**Comment by hach-que** [comment]
This change is incorrect and will regress the behaviour introduced in #88241.

With this change, the logic is now:

- Does `ConsoleUtils.EmitAnsiColorCodes` permit emitting color codes?
  - Yes, because `DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION` is set
  - Fallthrough to operating system check
- Is the operating system Windows?
  - Yes. Are we connected to a console?
    - No. Disallow colors.

The purpose of `DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION` is to override the detection done for Windows terminals, and to force ANSI colors even when output is redirected.

If you wanted to unify `ConsoleUtils`, I'd argue that the more correct change would be to replace `DoesConsoleSupportAnsi()` in it's entirety with returning the value from `ConsoleUtils.EmitAnsiColorCodes`, and then potentially update `ConsoleUtils.EmitAnsiColorCodes` to add the operating system check for Windows, something like this:

```csharp
/// <summary>Get whether to emit ANSI color codes.</summary>
public static bool EmitAnsiColorCodes
{
    get
    {
        // The flag starts at -1.  If it's no longer -1, it's 0 or 1 to represent false or true.
        int emitAnsiColorCodes = s_emitAnsiColorCodes;
        if (emitAnsiColorCodes != -1)
        {
            return Convert.ToBoolean(emitAnsiColorCodes);
        }

        // We've not yet computed whether to emit codes or not.  Do so now.  We may race with
        // other threads, and that's ok; this is idempotent unless someone is currently changing
        // the value of the relevant environment variables, in which case behavior here is undefined.

        // By default, we emit ANSI color codes if output isn't redirected, and suppress them if output is redirected.
        bool enabled = !Console.IsOutputRedirected;

        if (enabled)
        {
            // We subscribe to the informal standard from https://no-color.org/.  If we'd otherwise emit
            // ANSI color codes but the NO_COLOR environment variable is set, disable emitting them.
            enabled = Environment.GetEnvironmentVariable("NO_COLOR") is null;

            // On Windows, disallow colors if the terminal does not support them.
            if (enabled && OperatingSystem.IsWindows())
            {
                nint stdOutHandle = Interop.Kernel32.GetStdHandle(Interop.Kernel32.STD_OUTPUT_HANDLE);
                enabled = Interop.Kernel32.GetConsoleMode(stdOutHandle, out int consoleMode) &&
                    (consoleMode & Interop.Kernel32.ENABLE_VIRTUAL_TERMINAL_PROCESSING) != 0;
            }
        }
        else
        {
            // We also support overriding in the other direction.  If we'd otherwise avoid emitting color
            // codes but the DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION environment variable is
            // set to 1 or true, enable color.
            string? envVar = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION");
            enabled = envVar is not null && (envVar == "1" || envVar.Equals("true", StringComparison.OrdinalIgnoreCase));
        }

        // Store and return the computed answer.
        s_emitAnsiColorCodes = Convert.ToInt32(enabled);
        return enabled;
    }
}
```

**Comment by tarekgh** [comment]
@adamsitnik I am closing this PR for now and we can reopen when we decide about the behavior. Thanks!

**Pull request closed without merging**